File: app/macos/NativeStream/NativeStream/Views/Browser/ChannelCard.swift
import SwiftUI

// MARK: - Channel Card (UX-003)

struct ChannelCard: View {

    @Environment(EPGViewModel.self)      private var epgVM
    @Environment(PlayerViewModel.self)   private var playerVM
    @Environment(FavouritesManager.self) private var favourites

    let channel: Channel
    let onTap: () -> Void

    @State private var isHovered = false

    private var isLive: Bool    { epgVM.currentProgramme(for: channel) != nil }
    private var isPlaying: Bool { playerVM.currentChannel?.id == channel.id }
    private var programme: Programme? { epgVM.currentProgramme(for: channel) }

    private var borderColour: Color {
        isPlaying ? NS.accentBorder :
        isLive    ? Color(hex: "ef4444").opacity(0.157) :
                    NS.border
    }

    var body: some View {
        Button(action: onTap) {
            VStack(alignment: .leading, spacing: 6) {

                // Artwork
                ZStack(alignment: .bottom) {
                    ChannelLogoView(channel: channel, borderColour: borderColour)

                    if let prog = programme {
                        NSProgressBar(value: prog.progress, height: 3, glow: false)
                            .clipShape(.rect(
                                bottomLeadingRadius: NS.Radius.lg,
                                bottomTrailingRadius: NS.Radius.lg
                            ))
                    }

                    VStack {
                        HStack(alignment: .top) {
                            if isLive { NSLiveBadge(isLive: true) }
                            Spacer()
                            if isPlaying { playingBadge } else { starButton }
                        }
                        .padding(8)
                        Spacer()
                    }
                }
                .scaleEffect(isHovered ? 1.02 : 1.0)
                .animation(.easeOut(duration: 0.12), value: isHovered)

                // Channel name
                Text(channel.name)
                    .font(NS.Font.captionMed)
                    .foregroundStyle(NS.text)
                    .lineLimit(1)

                // EPG line
                if let prog = programme {
                    Text(prog.title)
                        .font(NS.Font.caption)
                        .foregroundStyle(NS.accent2)
                        .lineLimit(1)
                } else if let next = epgVM.nextProgramme(for: channel) {
                    Text(next.title)
                        .font(NS.Font.caption)
                        .foregroundStyle(NS.text3)
                        .lineLimit(1)
                }
            }
        }
        .buttonStyle(.plain)
        .onHover { isHovered = $0 }
    }

    private var playingBadge: some View {
        HStack(spacing: 3) {
            Image(systemName: "play.fill").font(.system(size: 7))
            Text("NOW").font(NS.Font.monoSm).fontWeight(.bold)
        }
        .foregroundStyle(.white)
        .padding(.horizontal, 6)
        .padding(.vertical, 3)
        .background(NS.accentGlow)
        .clipShape(RoundedRectangle(cornerRadius: NS.Radius.sm))
        .padding(8)
    }

    private var starButton: some View {
        Button { favourites.toggle(channel) } label: {
            Image(systemName: favourites.isFavourite(channel) ? "star.fill" : "star")
                .font(.system(size: 12))
                .foregroundStyle(favourites.isFavourite(channel) ? NS.amber : .white)
                .shadow(color: .black.opacity(0.5), radius: 3)
        }
        .buttonStyle(.plain)
        .padding(8)
    }
}

// MARK: - Channel Logo (UX-004)

struct ChannelLogoView: View {
    let channel: Channel
    var borderColour: Color = NS.border

    var body: some View {
        Group {
            if let url = channel.logoURL {
                AsyncImage(url: url) { phase in
                    switch phase {
                    case .success(let img):
                        img.resizable().scaledToFill()
                    default:
                        placeholder
                    }
                }
            } else {
                placeholder
            }
        }
        .aspectRatio(16/9, contentMode: .fit)
        .frame(maxWidth: .infinity)
        .background(NS.surface2)
        .clipShape(RoundedRectangle(cornerRadius: NS.Radius.lg))
        .overlay(
            RoundedRectangle(cornerRadius: NS.Radius.lg)
                .stroke(borderColour, lineWidth: 0.5)
        )
    }

    private var placeholder: some View {
        ZStack {
            NS.surface2
            Text(channel.name.prefix(3).uppercased())
                .font(NS.Font.label)
                .foregroundStyle(NS.text3)
        }
    }
}

File: app/android/app/src/main/java/com/nativestream/android/ui/screens/browse/ChannelCard.kt
// app/src/main/java/com/nativestream/android/ui/screens/browse/ChannelCard.kt
//
// Image-first card used in the Browse grid:
//   - 16:9 logo area via ChannelLogoView (Coil + initials fallback)
//   - LIVE badge top-left, star / ▶NOW badge top-right
//   - Progress bar bottom edge of artwork
//   - Playing state: accentBorder stroke + ▶NOW badge
//   - Ripple on press

package com.nativestream.android.ui.screens.browse

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.outlined.Star
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.res.vectorResource
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.nativestream.android.R
import com.nativestream.android.domain.model.Channel
import com.nativestream.android.domain.model.LiveEligibility
import com.nativestream.android.ui.components.NSLiveBadge
import com.nativestream.android.ui.components.NSProgressBar
import com.nativestream.android.ui.theme.NSColors
import com.nativestream.android.ui.theme.NSDimens
import com.nativestream.android.ui.theme.NSType
import com.nativestream.android.ui.viewmodel.EpgViewModel
import com.nativestream.android.ui.viewmodel.FavouritesViewModel
import com.nativestream.android.ui.viewmodel.PlayerViewModel

private val BADGE_H_PADDING = 6.dp
private val BADGE_V_PADDING = 3.dp
private val STAR_ICON_SIZE  = 12.dp
private val NOW_ICON_SIZE   = 7.dp

@Composable
fun ChannelCard(
    channel: Channel,
    playerViewModel: PlayerViewModel,
    epgViewModel: EpgViewModel,
    favouritesViewModel: FavouritesViewModel,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val dimens        = NSDimens.current
    val activeChannel by playerViewModel.activeChannel.collectAsState()
    val favourites    by favouritesViewModel.favouriteIds.collectAsState()

    val programme  = epgViewModel.currentProgramme(channel)
    val nextProg   = epgViewModel.nextProgramme(channel)
    val isPlaying  = activeChannel?.id == channel.id
    val isLive = LiveEligibility.isLive(channel, programme)
    val isFavourite = favourites.contains(channel.id)

    val borderColor = when {
        isPlaying -> NSColors.accentBorder
        isLive    -> NSColors.live.copy(alpha = 0.157f)
        else      -> NSColors.border
    }

    Column(
        verticalArrangement = Arrangement.spacedBy(6.dp),
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(dimens.radius.lg))
            .clickable(onClick = onClick),
    ) {
        // ── Artwork area ──────────────────────────────────────────────────────
        Box(modifier = Modifier.fillMaxWidth()) {
            ChannelLogoView(channel = channel, borderColor = borderColor)

            // Progress bar at bottom edge
            programme?.let { prog ->
                NSProgressBar(
                    value    = prog.progress.toFloat(),
                    modifier = Modifier
                        .fillMaxWidth()
                        .align(Alignment.BottomCenter),
                )
            }

            // Top overlay — LIVE badge + NOW/star
            Row(
                verticalAlignment = Alignment.Top,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(8.dp)
                    .align(Alignment.TopStart),
            ) {
                NSLiveBadge(isLive = isLive)
                Spacer(modifier = Modifier.weight(1f))
                if (isPlaying) {
                    NowBadge()
                } else {
                    StarButton(
                        isFavourite = isFavourite,
                        onClick     = { favouritesViewModel.toggle(channel) },
                    )
                }
            }
        }

        // ── Channel name ──────────────────────────────────────────────────────
        Text(
            text     = channel.name,
            style    = NSType.captionMedium(),
            color    = NSColors.text,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
        )

        // ── EPG line ──────────────────────────────────────────────────────────
        when {
            programme != null ->
                Text(
                    text     = programme.title,
                    style    = NSType.caption(),
                    color    = NSColors.accent2,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            nextProg != null ->
                Text(
                    text     = nextProg.title,
                    style    = NSType.caption(),
                    color    = NSColors.text3,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
        }
    }
}

// ── ▶NOW badge ────────────────────────────────────────────────────────────────

@Composable
private fun NowBadge() {
    val dimens = NSDimens.current
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(3.dp),
        modifier = Modifier
            .clip(RoundedCornerShape(dimens.radius.sm))
            .background(NSColors.accentGlow)
            .padding(horizontal = BADGE_H_PADDING, vertical = BADGE_V_PADDING),
    ) {
        Icon(
            imageVector        = ImageVector.vectorResource(R.drawable.ic_play),
            contentDescription = null,
            tint               = Color.White,
            modifier           = Modifier
                .height(NOW_ICON_SIZE)
                .width(NOW_ICON_SIZE),
        )
        Text(
            text   = "NOW",
            style  = NSType.monoSmall(),
            color  = Color.White,
        )
    }
}

// ── Star button ───────────────────────────────────────────────────────────────

@Composable
private fun StarButton(isFavourite: Boolean, onClick: () -> Unit) {
    Icon(
        imageVector        = if (isFavourite) Icons.Filled.Star else Icons.Outlined.Star,
        contentDescription = if (isFavourite) "Remove from favourites" else "Add to favourites",
        tint               = if (isFavourite) NSColors.amber else Color.White,
        modifier           = Modifier
            .height(STAR_ICON_SIZE)
            .width(STAR_ICON_SIZE)
            .clickable(onClick = onClick),
    )
}